core/src/apps/cardano/auxiliary_data.py

```diff
@@ -0,0 +1,252 @@
+
+
+def _validate_auxiliary_data_blob(auxiliary_data: bytes) -> None:
+    try:
+        # this also raises an error if there's some data remaining
+        cbor.decode(auxiliary_data)
+    except Exception:
+        raise INVALID_AUXILIARY_DATA
+
+
+def _validate_catalyst_registration_parameters(
+    catalyst_registration_parameters: Optional[
+        CardanoCatalystRegistrationParametersType
+    ],
+) -> None:
+    if not catalyst_registration_parameters:
+        raise INVALID_AUXILIARY_DATA
+
+    if len(catalyst_registration_parameters.voting_public_key) != PUBLIC_KEY_LENGTH:
+        raise INVALID_AUXILIARY_DATA
+
+    if not SCHEMA_STAKING_ANY_ACCOUNT.match(
+        catalyst_registration_parameters.staking_path
+    ):
+        raise INVALID_AUXILIARY_DATA
+
+    address_parameters = catalyst_registration_parameters.reward_address_parameters
+    if address_parameters.address_type == CardanoAddressType.BYRON:
+        raise INVALID_AUXILIARY_DATA
+
+
+async def show_auxiliary_data(
+    ctx: wire.Context,
+    keychain: seed.Keychain,
+    auxiliary_data: Optional[CardanoTxAuxiliaryDataType],
+    protocol_magic: int,
+    network_id: int,
+) -> None:
+    if not auxiliary_data:
+        return
+
+    if (
+        auxiliary_data.metadata
+        and auxiliary_data.metadata.type == CardanoMetadataType.CATALYST_REGISTRATION
+    ):
+        # ensured by _validate_auxiliary_data
+        assert auxiliary_data.metadata.catalyst_registration_parameters is not None
+
+        await _show_catalyst_registration(
+            ctx,
+            keychain,
+            auxiliary_data.metadata.catalyst_registration_parameters,
+            protocol_magic,
+            network_id,
+        )
+
+    auxiliary_data_bytes = get_auxiliary_data_bytes(
+        keychain, auxiliary_data, protocol_magic, network_id
+    )
+    # get_auxiliary_data_bytes returns none only if auxiliary_data is None, which is checked above
+    assert auxiliary_data_bytes is not None
+
+    auxiliary_data_hash = hash_auxiliary_data(bytes(auxiliary_data_bytes))
+    await show_auxiliary_data_hash(ctx, auxiliary_data_hash)
+
+
+async def _show_catalyst_registration(
+    ctx: wire.Context,
+    keychain: seed.Keychain,
+    catalyst_registration_parameters: CardanoCatalystRegistrationParametersType,
+    protocol_magic: int,
+    network_id: int,
+) -> None:
+    public_key = catalyst_registration_parameters.voting_public_key
+    encoded_public_key = bech32.encode(HRP_JORMUN_PUBLIC_KEY, public_key)
+    staking_path = catalyst_registration_parameters.staking_path
+    reward_address = derive_human_readable_address(
+        keychain,
+        catalyst_registration_parameters.reward_address_parameters,
+        protocol_magic,
+        network_id,
+    )
+    nonce = catalyst_registration_parameters.nonce
+
+    await confirm_catalyst_registration(
+        ctx, encoded_public_key, staking_path, reward_address, nonce
+    )
+
+
+def get_auxiliary_data_bytes(
```

Comment:
`get_auxiliary_data_cbor` would be more descriptive to make it clear what kind of "bytes" we are getting here, which is otherwise obvious from the type annotation

Reply:
https://github.com/vacuumlabs/trezor-firmware/pull/57/commits/f73f563cd7b142506f75b45773fe521898ae04ab